web/web.go

```diff
@@ -42,25 +42,28 @@ func (me *HtmlTemplatePage) Render(c *api.Context, w http.ResponseWriter) {
 func InitWeb() {
 	l4g.Debug("Initializing web routes")
 
+	mainrouter := api.Srv.Router
+
 	staticDir := utils.FindDir("web/static")
 	l4g.Debug("Using static directory at %v", staticDir)
-	api.Srv.Router.PathPrefix("/static/").Handler(http.StripPrefix("/static/",
-		http.FileServer(http.Dir(staticDir))))
-
-	api.Srv.Router.Handle("/", api.AppHandler(root)).Methods("GET")
-	api.Srv.Router.Handle("/login", api.AppHandler(login)).Methods("GET")
-	api.Srv.Router.Handle("/signup_team_confirm/", api.AppHandler(signupTeamConfirm)).Methods("GET")
-	api.Srv.Router.Handle("/signup_team_complete/", api.AppHandler(signupTeamComplete)).Methods("GET")
-	api.Srv.Router.Handle("/signup_user_complete/", api.AppHandler(signupUserComplete)).Methods("GET")
-
-	api.Srv.Router.Handle("/logout", api.AppHandler(logout)).Methods("GET")
-
-	api.Srv.Router.Handle("/verify", api.AppHandler(verifyEmail)).Methods("GET")
-	api.Srv.Router.Handle("/find_team", api.AppHandler(findTeam)).Methods("GET")
-	api.Srv.Router.Handle("/reset_password", api.AppHandler(resetPassword)).Methods("GET")
-
-	csr := api.Srv.Router.PathPrefix("/channels").Subrouter()
-	csr.Handle("/{name:[A-Za-z0-9-]+(__)?[A-Za-z0-9-]+}", api.UserRequired(getChannel)).Methods("GET")
+	mainrouter.PathPrefix("/static/").Handler(http.StripPrefix("/static/", http.FileServer(http.Dir(staticDir))))
+
+	mainrouter.Handle("/", api.AppHandlerIndependent(root)).Methods("GET")
+	mainrouter.Handle("/{team:[A-Za-z0-9-]+(__)?[A-Za-z0-9-]+}", api.AppHandler(login)).Methods("GET")
+	mainrouter.Handle("/{team:[A-Za-z0-9-]+(__)?[A-Za-z0-9-]+}/", api.AppHandler(login)).Methods("GET")
+	mainrouter.Handle("/{team:[A-Za-z0-9-]+(__)?[A-Za-z0-9-]+}/login", api.AppHandler(login)).Methods("GET")
+	mainrouter.Handle("/{team:[A-Za-z0-9-]+(__)?[A-Za-z0-9-]+}/logout", api.AppHandler(logout)).Methods("GET")
+	mainrouter.Handle("/{team:[A-Za-z0-9-]+(__)?[A-Za-z0-9-]+}/reset_password", api.AppHandler(resetPassword)).Methods("GET")
+	// Bug in gorilla.mux pervents us from using regex here.
+	mainrouter.Handle("/{team}/channels/{channelname}", api.UserRequired(getChannel)).Methods("GET")
+
+	// Anything added here must have an _ in it so it does not conflict with team names
```

Comment:
I wonder if these should be moved above the handlers with "{team:" stuff just in case.  I think first route wins?  vs depending on _.  my fear is someone will change the code without understanding the subtly. (0/5)